velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1528,6 +1528,30 @@ struct CurrentDateFunction {
   }
 };
 
+template <typename T>
+struct CurrentTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  int64_t result_{0};
+  const tz::TimeZone* timeZone_ = nullptr;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /* type */,
+      const core::QueryConfig& config) {
+    Timestamp ts = Timestamp::fromMillis(config.sessionStartTimeMs());
+    timeZone_ = getTimeZoneFromConfig(config);
+    if (timeZone_ != nullptr) {
+      result_ = pack(ts, timeZone_->id());
+      return;
+    }
+    result_ = pack(ts, Timestamp::defaultTimezone().id());
```

Comment:
@kgpai : I feel we should error here instead of pick up the default timezone. We always want the value on the co-ordinator and not the local timezone.
wdyt ?

Reply:
Done. @aditi-pandit Please take another look. Thanks.